ord_schema/interface/query.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2020 Open Reaction Database Project Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library for executing PostgreSQL queries on the ORD."""
+
+import binascii
+
+from absl import logging
+import psycopg2
+
+from ord_schema.proto import dataset_pb2
+from ord_schema.proto import reaction_pb2
+
+
+class OrdPostgres:
+    """Class for performing SQL queries on the ORD."""
+    def __init__(self,
+                 dbname='ord',
+                 user='postgres',
+                 password='postgres',
+                 host=None,
+                 port=None):
+        """Initializes an instance of OrdPostgres.
+
+        Args:
+            dbname: Text database name.
+            user: Text user name.
+            password: Text user password.
+            host: Text host name.
+            port: Integer port.
+        """
+        self._dbname = dbname
+        self._user = user
+        self._password = password
+        self._host = host
+        self._port = port
+        self._db = None
+
+    def __enter__(self):
+        self._db = psycopg2.connect(dbname=self._dbname,
+                                    user=self._user,
+                                    password=self._password,
+                                    host=self._host,
+                                    port=self._port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        del exc_type, exc_val, exc_tb  # Unused.
+        self._db.close()
+
+    def get_cursor(self):
+        return Cursor(self._db)
+
+    def substructure_search(self, smiles, table, limit=100):
```

Comment:
It would be nice to have a substructure search using SMARTS queries too. Perhaps the easiest thing to do is generalize the `smiles` arg to `query` and add a `smarts=False` kwarg. It looks like the only change in the postgres query is appending "::qmol" to the query string outside of quotation marks

Reply:
Done.